fbw-a32nx/src/wasm/systems/a320_systems/src/payload/mod.rs

```diff
@@ -0,0 +1,663 @@
+use enum_map::{Enum, EnumMap};
+use lazy_static::lazy_static;
+
+use std::{cell::Cell, rc::Rc, time::Duration};
+
+use uom::si::{f64::Mass, mass::kilogram};
+
+use systems::{
+    payload::{BoardingRate, Cargo, CargoInfo, GsxState, Pax, PaxInfo},
+    simulation::{
+        InitContext, Read, SimulationElement, SimulationElementVisitor, SimulatorReader,
+        SimulatorWriter, UpdateContext, VariableIdentifier, Write,
+    },
+};
+
+#[cfg(test)]
+pub mod test;
+
+#[derive(Debug, Clone, Copy, Enum)]
+pub enum A320Pax {
+    A,
+    B,
+    C,
+    D,
+}
+impl A320Pax {
+    pub fn iterator() -> impl Iterator<Item = A320Pax> {
+        [A320Pax::A, A320Pax::B, A320Pax::C, A320Pax::D]
+            .iter()
+            .copied()
+    }
+}
+
+#[derive(Debug, Clone, Copy, Enum)]
+pub enum A320Cargo {
+    FwdBaggage,
+    AftContainer,
+    AftBaggage,
+    AftBulkLoose,
+}
+impl A320Cargo {
+    pub fn iterator() -> impl Iterator<Item = A320Cargo> {
+        [
+            A320Cargo::FwdBaggage,
+            A320Cargo::AftContainer,
+            A320Cargo::AftBaggage,
+            A320Cargo::AftBulkLoose,
+        ]
+        .iter()
+        .copied()
+    }
+}
+
+lazy_static! {
+    static ref A320_PAX: EnumMap<A320Pax, PaxInfo> = EnumMap::from_array([
+        PaxInfo::new(36, "PAX_A", "PAYLOAD_STATION_1_REQ",),
+        PaxInfo::new(42, "PAX_B", "PAYLOAD_STATION_2_REQ",),
+        PaxInfo::new(48, "PAX_C", "PAYLOAD_STATION_3_REQ",),
+        PaxInfo::new(48, "PAX_D", "PAYLOAD_STATION_4_REQ",)
+    ]);
+    static ref A320_CARGO: EnumMap<A320Cargo, CargoInfo> = EnumMap::from_array([
+        CargoInfo::new(
+            Mass::new::<kilogram>(3402.),
+            "CARGO_FWD_BAGGAGE_CONTAINER",
+            "PAYLOAD_STATION_5_REQ",
+        ),
+        CargoInfo::new(
+            Mass::new::<kilogram>(2426.),
+            "CARGO_AFT_CONTAINER",
+            "PAYLOAD_STATION_6_REQ",
+        ),
+        CargoInfo::new(
+            Mass::new::<kilogram>(2110.),
+            "CARGO_AFT_BAGGAGE",
+            "PAYLOAD_STATION_7_REQ",
+        ),
+        CargoInfo::new(
+            Mass::new::<kilogram>(1497.),
+            "CARGO_AFT_BULK_LOOSE",
+            "PAYLOAD_STATION_8_REQ",
+        )
+    ]);
+}
+
```

Comment:
I don't really think you need this. You can `impl Display` for the `A320Pax` and `A320Cargo` enums to later construct the VariableIdentifiers.

Suggestion: 

impl Display (you'll need another one for `A320Cargo`)

```suggestion
impl Display for A320Pax {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            A320Pax::A => write!(f, "PAX_A"),
            A320Pax::B => write!(f, "PAX_B"),
            A320Pax::C => write!(f, "PAX_C"),
            A320Pax::D => write!(f, "PAX_D"),
        }
    }
}

```

You will have to deal with the max_pax somewhere else. Either in the enum itself by storing the value, or in `Pax`. If you do this, you might be able to remove `PaxInfo` and `CargoInfo` all together.